ethcore/types/src/state_diff.rs

```diff
@@ -19,28 +19,36 @@
 use std::fmt;
 use std::ops::*;
 use std::collections::BTreeMap;
-use ethereum_types::Address;
+use ethereum_types::{Address,H256};
 use account_diff::*;
 
 /// Expression for the delta between two system states. Encoded the
 /// delta of every altered account.
 #[derive(Debug, PartialEq, Eq, Clone)]
 pub struct StateDiff {
 	/// Raw diff key-value
-	pub raw: BTreeMap<Address, AccountDiff>
+	pub raw: BTreeMap<Address, AccountDiff>,
+	pub transaction_hash: H256
```

Comment:
`transaction_hash` is not a part of a `StateDiff` and should not be a property of a `struct StateDiff`